haproxy/haproxy.cfg

```diff
@@ -10,20 +10,28 @@ global
 
 frontend http
   bind *:80
-  redirect scheme https code 301
+
+  acl acme_path path_beg -i /.well-known/acme-challenge
+  use_backend acme if acme_path
+
+  redirect scheme https code 301 unless acme_path
 
 frontend https
-  bind *:443 ssl crt /certs
+  bind *:443 ssl crt /certs/certs
 
   default_backend foodsoft_web
+
   acl sharedlists_path path_beg /sharedlists
   use_backend sharedlists_web if sharedlists_path
 
   http-response set-header Strict-Transport-Security "max-age=16000000;"
+  http-response set-header X-Frame-Options "DENY"
```

Comment:
I like this option, but I don't think it's a good idea to set it here. I'd move it into the corresponding backends (foodsoft, sharedlists, ...)